File: lintcode/Remove_Element.cpp
class Solution {
public:
    /** 
     *@param A: A list of integers
     *@param elem: An integer
     *@return: The new length after remove
     */
    int removeElement(vector<int> &A, int elem) {
        if(A.empty()) return 0;
        
        int i = 0;
        
        for(int j = 0; j < A.size(); j++){
            if(A[j] != elem) A[i++] = A[j];
        }
        
        return i;
    }
};

File: leetcode/java/RemoveElement.java
class Solution {
    public int removeElement(int[] nums, int val) {
        if(nums == null) throw new IllegalArgumentException("x");
        int i = 0;
        int j = 0;
        while(j < nums.length){
            if(nums[j] == val) j++;
            else nums[i++] = nums[j++];
        }
        return i;
    }
}
